qemu/tests/qemu_disk_img_resize.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from autotest.client.shared import error
+from qemu.tests import qemu_disk_img
+
+
+class ResizeTest(qemu_disk_img.QemuImgTest):
+
+    def __init__(self, test, params, env):
+        self.tag = params.get("image_resize", "image1")
+        t_params = params.object_params(self.tag)
+        super(ResizeTest, self).__init__(test, t_params, env, self.tag)
+
+    def check_size(self, image_size, virtual_size):
+        """
+        Check the disk image size as if it had been change with resize;
+        """
+        re_size = self.params.get("re_size")
+        human = {'B': 1,
+                 'K': 1024,
+                 'M': 1048576,
+                 'G': 1073741824,
+                 }
+        if human.has_key(re_size[-1]):
+            size = int(re_size[:-1]) * human[re_size[-1]]
+            if "+" in re_size[0]:
+                if virtual_size != image_size + size:
+                    raise error.TestError("Check the image size failed!")
+            elif "-" in re_size[0]:
+                if virtual_size != image_size - size:
+                    raise error.TestError("Check the image size failed!")
+            else:
+                if virtual_size != size:
+                    raise error.TestError("Check the image size failed!")
+        else:
+            raise error.TestError("Invalid re_size: %s" % re_size)
+
+
+def run(test, params, env):
+    """
+    'qemu-img' resize functions test:
+
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
```

Comment:
Please doc your test steps in doc string and print it via error.context(xxx, logging.info)